sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala

```diff
@@ -497,6 +497,17 @@ class SQLMetricsSuite extends SparkFunSuite with SQLMetricsTestUtils with Shared
     }
   }
 
+  test("SPARK-25278: output metrics are wrong for plans repeated in the query") {
+    val name = "demo_view"
+    sql(s"CREATE OR REPLACE VIEW $name AS VALUES 1,2")
+    val view = spark.table(name)
+    val union = view.union(view)
+    testSparkPlanMetrics(union, 1, Map(
+      0L -> ("Union" -> Map()),
+      1L -> ("LocalTableScan" -> Map("number of output rows" -> 2L)),
+      2L -> ("LocalTableScan" -> Map("number of output rows" -> 2L))))
+  }
+
```

Comment:
In addition to this end-to-end test, can we add fine-grained tests for the scenario you described in the PR description?

Reply:
yes, I am adding a test to the `PlannerSuite` which ensures that plans are different instances. Thanks.